mempool/src/pool/store.rs

```diff
@@ -0,0 +1,397 @@
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use std::collections::HashMap;
+
+use common::chain::tokens::OutputValue;
+use common::chain::transaction::Transaction;
+use common::chain::OutPoint;
+use common::primitives::amount::Amount;
+use common::primitives::id::WithId;
+use common::primitives::Id;
+use common::primitives::Idable;
+use common::primitives::H256;
+
+use logging::log;
+
+use utils::newtype;
+
+use crate::error::Error;
+use crate::error::TxValidationError;
+
+use super::Ancestors;
+use super::Conflicts;
+use super::Descendants;
+use super::Time;
+
+newtype! {
+    #[derive(Debug, PartialEq, Eq, Ord, PartialOrd)]
+    pub(super) struct DescendantScore(Amount);
+}
+
+#[derive(Debug)]
+pub(super) struct MempoolStore {
```

Comment:
You're right. Not sure what the advantage is in making this just pub, though?

Reply:
> You're right. Not sure what the advantage is in making this just pub, though?

That's a good question. There is a difference in what happens if the modules are moved around or publicity of the parent changes. Not quite sure what's better but maybe `pub(super)` my actually be preferable as it's tighter.

Reply:
Usually using pub(super) or similar is considered (with a mild exaggeration) a design flaw because ideally, everything that can be extracted should go into a separate crate, but that's not real life, as it's impractical. The reason why pub(super) is bad in this context is because it signals that "this module is accessible to others that can see it but shouldn't touch it", but that isn't the case. So it's bad in the sense "don't add qualifiers that aren't necessary". In fact, not only that this signals a false positive, but also hides design flaws, because this module could be imported with pub mod, and no one would notice because it defeated the pub purpose.